docs/proposals/20210524-managed-external-etcd.md

```diff
@@ -0,0 +1,570 @@
+    serviceDomain: "cluster.local"
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+    kind: DockerCluster
+    name: "my-cluster"
+    namespace: "default"
+  controlPlaneRef:
+    kind: KubeadmControlPlane
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
+    name: "my-cluster-control-plane"
+    namespace: "default"
+  managedExternalEtcdRef:
+    kind: EtcdadmCluster
+    apiVersion: etcd.cluster.x-k8s.io/v1alpha4
+    name: "my-cluster-etcd-cluster"
+    namespace: "default"
+---
+kind: EtcdadmCluster
+apiVersion: etcd.cluster.x-k8s.io/v1alpha4
+metadata:
+  name: "my-cluster-etcd-cluster"
+  namespace: default
+spec:
+  etcdadmConfigSpec:
+    version: 3.4.13 #optional
+    # etcdadmArgs:
+    # users:
+  replicas: 3
+  infrastructureTemplate:
+    kind: DockerMachineTemplate
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+    name: "etcd-plane"
+    namespace: "default"
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: DockerCluster
+metadata:
+  name: "my-cluster"
+  namespace: "default"
+---
+kind: KubeadmControlPlane
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
+metadata:
+  name: "abcd-control-plane"
+  namespace: "default"
+spec:
+  replicas: 3
+  infrastructureTemplate:
+    kind: DockerMachineTemplate
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+    name: "abcd-control-plane"
+    namespace: "default"
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      controllerManager:
+        extraArgs: {enable-hostpath-provisioner: 'true'}
+      apiServer:
+        certSANs: [localhost, 127.0.0.1]
+      etcd:
+        external:
+          endpoints: []
+          caFile: "/etc/kubernetes/pki/etcd/ca.crt"
+          certFile: "/etc/kubernetes/pki/apiserver-etcd-client.crt"
+          keyFile: "/etc/kubernetes/pki/apiserver-etcd-client.key"
+    initConfiguration:
+      nodeRegistration:
+        criSocket: /var/run/containerd/containerd.sock
+        kubeletExtraArgs: {eviction-hard: 'nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%'}
+    joinConfiguration:
+      nodeRegistration:
+        criSocket: /var/run/containerd/containerd.sock
+        kubeletExtraArgs: {eviction-hard: 'nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%'}
+  version: "v1.20.2"
+```
+
+![etcdcluster-sequence](images/managed-etcd/etcdcluster-sequence.png)
+
+- Cluster.Spec will have a new field `ManagedExternalEtcdRef` of type `ObjectReference` (same as `ControlPlaneRef`). For using CAPI managed etcd cluster, user will set this field in Cluster manifest as shown above.
+- Cluster controller's `reconcileControlPlane` will check if `managedExternalEtcdRef` is set on cluster spec and if External etcd cluster does not have `CreationComplete` set to true, it will pause the control plane provisioning by setting clusterv1.Paused annotation (`cluster.x-k8s.io/paused`) on it. This will allow etcd cluster provisioning to occur first.
```

Comment:
We should avoid using pause as a way to synchronise controllers; pause is for maintenance or exceptional operations like move.